src/domain/account.ts

```diff
@@ -1,264 +1,318 @@
+    networks?: NetworkString[]
+  ): Promise<AccountFactory> {
+    const factory = new AccountFactory(walletRepository);
+    for (const net of networks || (['liquid', 'testnet', 'regtest'] as NetworkString[])) {
+      const chainSource = await appRepository.getChainSource(net);
+      if (!chainSource) {
+        console.error('Chain source not found', net);
+        continue;
+      }
+      factory.chainSources.set(net, chainSource);
+    }
+
+    const masterBlindingKey = await walletRepository.getMasterBlindingKey();
+    if (!masterBlindingKey) throw new Error('Master blinding key not found');
+    factory.masterBlindingKey = masterBlindingKey;
+    return factory;
+  }
+
+  async make(network: NetworkString, accountName: string): Promise<Account> {
+    const { [accountName]: details } = await this.walletRepository.getAccountDetails(accountName);
+    if (!details) throw new Error('Account not found');
+    if (details.accountNetworks.indexOf(network) === -1)
+      throw new Error(`Account ${accountName} does not support network: ${network}`);
+    const chainSource = this.chainSources.get(network);
+    if (!chainSource) throw new Error('Chain source not found');
+    return new Account({
+      name: accountName,
+      chainSource,
+      masterPublicKey: details.masterPublicKey,
+      masterBlindingKey: this.masterBlindingKey!,
+      walletRepository: this.walletRepository,
+      network,
+    });
+  }
 }
 
-// Factory for mainAccount
-function createMainAccount(
-  encryptedMnemonic: EncryptedMnemonic,
-  data: MnemonicAccountData
-): MnemonicAccount {
-  return {
-    type: AccountType.MainAccount,
-    getSigningIdentity: (password: string, network: NetworkString) =>
-      restoredMnemonic(decrypt(encryptedMnemonic, password), data.restorerOpts[network], network),
-    getWatchIdentity: (network: NetworkString) =>
-      restoredMasterPublicKey(
-        data.masterXPub,
-        data.masterBlindingKey,
-        data.restorerOpts[network],
-        network
-      ),
-    getNotRestoredWatchIdentity: (network: NetworkString) =>
-      newMasterPublicKey(data.masterXPub, data.masterBlindingKey, network),
-    getDeepRestorer: (network: NetworkString) => {
-      const pubkey = newMasterPublicKey(data.masterXPub, data.masterBlindingKey, network);
-      return makeRestorerFromChainAPI<MasterPublicKey>(
-        pubkey,
-        (isChange, index) => pubkey.getAddress(isChange, index).address.confidentialAddress
+// Account is a readonly way to interact with the account data (transactions, utxos, scripts, etc.)
+export class Account {
+  private chainSource: ChainSource;
+  private network: networks.Network;
+  private node: BIP32Interface;
+  private blindingKeyNode: Slip77Interface;
+  private walletRepository: WalletRepository;
+  readonly name: string;
+
+  static BASE_DERIVATION_PATH = "m/84'/1776'/0'";
+  static BASE_DERIVATION_PATH_LEGACY = "m/84'/0'/0'";
+  static BASE_DERIVATION_PATH_TESTNET = "m/84'/1'/0'";
```

Comment:
In LDK (so current apps), the default base derivation path is the same for testnet and mainnet. Not anymore.
What is the reasoning to change that?

Reply:
To be compatible with Green